challenges/platform/README.md

```diff
@@ -0,0 +1,59 @@
+
+
+# Rationale
+
+These exercises have two goals.
+
+- It helps us understand what to expect from you as an engineer, how you reason about production systems and investigate faults.
+- It helps you get a feel for what it would be like to work at Teleport, as this exercise aims to simulate some of our day-as-usual and expose you to the type of work we're doing here.
+
+We believe this technique is better and is more fun compared to whiteboard/quiz interviews so common in the industry. It's not without the downsides as it could take longer than traditional interviews. [Some of the best teams use coding challenges.](https://sockpuppet.org/blog/2015/03/06/the-hiring-post/)
+
+We appreciate your time and are looking forward to hacking on this challenge together.
+
+
+# Backgrounds
+
+## Systems / Backend Developers
+Systems developers on the platform team leverage a strong background in software development to build, maintain, deploy core production services.
+
+### Challenge
+1. [Systems developer challenge](../systems/worker.pdf)
+
+### Levels
+See [levels.pdf](../../levels.pdf)
+
+
+## Devops / DevSecOps / System Administrators / Production engineers / etc
+Team members with experience in various operations focused disciplines are focussed on the day to day delivery of platform components, such as build/testing automation of infrastructure as code, deploying and configuring our metrics, alarming and logging systems, automating build pipeliness, and other day to day activities to enable our cloud infrastructure.
+
+
+### Challenge
+We use two interviews for team members with operational backgrounds, the first is a lightweight scripting / programming challenge, and the second is a troubleshooting production test. See the following links for outlines of each challenge.
+1. [Devops development challenge](devops.md)
+2. [Troubleshooting challenge](troubleshooting.md)
+
+### Levels
+
+
+| Level  | Oncall  | Operations  | Security  | Communications / Writing  | Networking  | Systems Engineering  | OS Understanding  | Tooling |
+|---|---|---|---|---|---|---|---|---|
+| 1 | Participates in on-call and able to execute run books | Documents and Executes maintenance activities in production | Able to follow security guidelines and best practices in a production environment  | Contributor to internal documentation such as updating run books | Able to troubleshoot introductory networking issues, network policies, routing, etc. | Focus on and contribute to specific area of production system (such as the deployment of our monitoring stack)  | Working understanding of core OS concepts, filesystem, networking, cgroups, namespaces, security mechanisms, etc.  | Work with team members on deployment and maintenance of internal tooling |
```

Comment:
Do we want to have our entry level participate in on-call?  I'd argue we should have on-call start at l2 -- with the idea that entry level may have 0 industry experience, and thus probably need at least 6m-1y of experience and shadowing before they'd be ready to participate as a first class member.

I say this because I don't want our entry level to actually be "1-2 years of experience"  -- it stops being entry level -- and l2 on-call also gives us a carrot to entice entry level towards on-call.

Under the theory of "you need to perform at the level before you're promoted to it", the upper end of l1 can start participating in the rotation in preparation for promotion to l2.

Language suggestion:

l1 Shadows on-call and executes run books
l2 Serves as primary on call, able to investigate and solve many production incidents with peer review

Reply:
So my line of thinking when writing this, is that a new grad / entry level would be expected to be oncall relatively quickly, with backup / escalation to a senior engineer. So if there is a prescribed runbook, they can just execute it and get experience. For current state we might have too many risks in our architecture to really throw a new grad the keys to prod, but eventually we'll get to the state... unless we get to the point where we need to double up to get admin on prod.

This is how I got started, I was oncall for a key piece of network infra within a couple months of graduating.

Reply:
I think we're saying the same thing with different language.  My "shadow" is your "oncall with backup".  The main distinction for me is: can I take a hike without my phone during the shift?  If the answer is "no" then the mentee is still in some phase of learning, and not full on-call.

Reply:
Perhaps:

l1:
> Participates in on-call with backup...